chapters/en/slides/module7/module7_04.Rmd

```diff
@@ -0,0 +1,342 @@
+
+Notes:
+Now that we know how Altair creates charts,
+we can also understand why there is an issue with large datasets.
+
+You can see the JSON chart spec printed in this slide again.
+Note that the data here is just a string value pointing to a URL
+because we used `data.cars.url` 
+to create the chart.
+
+If we instead of a URL would use a pandas dataframe (e.g. `data.cars()`)
+**all** the data would be included in the chart specification.
+In the JSON format,
+this is almost 4500 lines of code!
+There is not enough room to show that in this slide,
+but you can click the link under the code
+to see an example of what it would look like.
+
+---
+
+## Including large data sets in chart
+
+```python
+# Disable the warning
+alt.data_transformers.disable_max_rows()
+```
+
+[More details about how to work with large datasets can be found in the docs](https://altair-viz.github.io/user_guide/faq.html#maxrowserror-how-can-i-plot-large-datasets)
+
+Notes:
+Including all the data is great for reproducibility
+and it works well with small to medium sized data.
+However,
+with large datasets
+the size of each chart would become really big,
+and with many charts in the same notebook
+the overall file size would be huge.
+
+To avoid this happening by mistake
+Altair shows a warning when you try to plot a dataset 
+with more than 5000 rows.
+If you are sure you want to plot a dataframe 
+with more rows than that,
+you can disable the warning with the line of code on the top of this slide.
+
+However,
+your plots would still be very big.
+If you would like to avoid creating big plots,
+you can use a URL as we did in the first slide and in module three.
+
+An URL is not always an option,
+and it is not convenient
+if you need to do some data wrangling with pandas first.
```

Comment:
```suggestion
Including all the data is great for reproducibility
and it works well with small to medium sized data.
However,
with large datasets
the size of each chart would become really big,
and with many charts in the same notebook
the overall file size would be huge.

To avoid this happening by mistake
Altair shows a warning when you try to plot a dataset 
with more than 5000 rows.
If you are sure you want to  create plot using a dataframe 
with more rows than that,
you can disable the warning by typing: `alt.data_transformers.disable_max_rows()`.

However, as discussed in the last slide,
your charts would still be very big because the data would be included in the spec for the chart.
If you would like to avoid creating big plots,
you can use a URL as we did in the first slide and in module three.

An URL is not always an option,
and it is not convenient
if you need to do some data wrangling with pandas first. So we will next
discuss another way to manage this challenge.
```